adapters/adapter.go

```diff
@@ -9,32 +9,42 @@ import (
 	"time"
 )
 
+// Adapters connect prebid-server to a demand partner. Their primary purpose is to produce bids
+// in response to Auction requests.
 type Adapter interface {
+	// A name which uniquely identifies this adapter.
+	// This must not overlap with any other adapters in prebid-server.
 	Name() string
 	FamilyName() string
 	GetUsersyncInfo() *pbs.UsersyncInfo
+	// Produce bids which should be considered, given the auction params.
```

Comment:
I don't think we need to limit flexibility; it's more just making clear where the callouts happen (assuming they do).